editor/modes/visual-editor/block-list.js

```diff
@@ -65,21 +65,24 @@ class VisualEditorBlockList extends Component {
 		this.lastClientY = clientY;
 	}
 
-	setBlockRef( ref, uid ) {
+	setBlockRef( ref ) {
+		const uid = ref.props.uid;
```

Comment:
The only thing that bothers me about this, is that it's not obvious that we need to change this if we rename/remove the `uid` prop (or if React changes the element object shape)

Reply:
>The only thing that bothers me about this, is that it's not obvious that we need to change this if we rename/remove the `uid` prop (or if React changes the element object shape)

Sure, I can see this. I imagine if we did change the `uid` prop, there'd be much more impacted code than just what's referenced here. I don't love reaching into the element props, but don't necessarily see it as an unstable API.

How could this be improved? Would a comment help clarify?

I had considered removing the `ref` entirely and passing the node from the block component itself when selection starts, but this isn't sufficient since the block list needs to compare the selection with _all_ rects of blocks in the list, so it needs access to every block's DOM node.

Reply:
Let's add a comment. Anyway nice improvement 👍